pm.pl

```diff
@@ -1,25 +1,35 @@
 :- use_module(download).
 :- use_module(config).
-
+:- use_module(find_modules).
 create_dir_if_needed :- \+ exists_directory("deps"),
      make_directory("deps").
      
-download_to_disk(URL, Version) :- 
+download_to_disk(URL, Version, PackageName) :- 
     split_string(URL, "/", "", Pieces),
     last(Pieces, PackageName),
     format(atom(Formatted), 'deps/~w', [PackageName]),
     down(URL, Formatted).   
 
 sync :- create_dir_if_needed,
-        forall(dependency(URL, Version), download_to_disk(URL, Version)).
+        forall(dependency(URL, Version), download_to_disk(URL, Version, _)).
+
+handle_module_sources(File) :-
+                               format(atom(Formatted), 'deps/~w', [File]),
+                               find_deps(Formatted, [(Module, _)]),
+                               open('deps/sources.pl', append, Handle),
```

Comment:
in general, use https://www.swi-prolog.org/pldoc/doc_for?object=setup_call_cleanup/3 to wrap the open/close calls

Reply:
This is something we can refactor later though so you don't necessarily need to change it now